src/lib/images/proxy.ts

```diff
@@ -152,7 +152,32 @@ export const initializeProxy = async function ({
       console.log(`fullImageUrl: ${fullImageUrl}`)
       req.url = `/${modifiers}/${fullImageUrl}`
     } else {
-      // If the image is remote, we can just pass the URL as is
+      // If the image is remote, we first check if it's allowed by any of patterns
+      if (!remoteImages.some((remoteImage) => remoteImage.test(sourceImagePath))) {
+        const remoteImageNotAllowedLogMessage = `Remote image "${sourceImagePath}" source for Image CDN is not allowed.`
+
+        // Contextual information about the remote image configuration is throttled
+        // to avoid spamming the console as it's quite verbose
+        // Each not allowed remote image will still be logged, just without configuration details
+        if (Date.now() - lastTimeRemoteImagesConfigurationDetailsMessageWasLogged > 1000 * 30) {
+          console.error(
+            `${remoteImageNotAllowedLogMessage}\n\n${
+              remoteImages.length === 0
+                ? 'Currently no remote images are allowed.'
+                : `Currently allowed remote images configuration details:\n${remoteImages
+                    .map((pattern) => ` - ${pattern}`)
+                    .join('\n')}`
+            }\n\nSee https://docs.netlify.com/image-cdn/overview/#remote-path how to configure allowed remote images.`,
+          )
+          lastTimeRemoteImagesConfigurationDetailsMessageWasLogged = Date.now()
+        } else {
+          console.error(remoteImageNotAllowedLogMessage)
+        }
```

Comment:
This is what this does in practice (this was refreshing same page multiple times to trigger the not allowed requests): 
![image](https://github.com/netlify/cli/assets/419821/f1adfabe-6011-4296-973b-0801572d9c48)

The more verbose variant try to help guide user why things happen to allow faster diagnose and fixing time by pointing to relevant resource (not sure if it should be path like that or some link shortener could be used, or even if we are fine with adding links that might get out of sync all together). The more verbose variant is ... verbose so that's why I only display it at most once per 30 seconds in current shape of things, otherwise log reading (when combined with logs produced by other things) becomes very messy frustrating for users